packages/flutter_tools/templates/module/ios/library/Flutter.tmpl/podhelper.rb.tmpl

```diff
@@ -118,30 +80,24 @@ end
 #                                          Optional, defaults to two levels up from the directory of this script.
 #                                          MyApp/my_flutter/.ios/Flutter/../..
 def install_flutter_application_pod(flutter_application_path)
-  current_directory_pathname = Pathname.new File.expand_path('..', __FILE__)
-  app_framework_dir = File.expand_path('App.framework', current_directory_pathname.to_path)
-  app_framework_dylib = File.join(app_framework_dir, 'App')
-  if !File.exist?(app_framework_dylib)
-    # Fake an App.framework to have something to link against if the xcode backend script has not run yet.
-    # CocoaPods will not embed the framework on pod install (before any build phases can run) if the dylib does not exist.
-    # Create a dummy dylib.
-    FileUtils.mkdir_p(app_framework_dir)
-    sdk_path = `xcrun --sdk iphoneos --show-sdk-path`.strip
-    `echo "static const int Moo = 88;" | xcrun clang -x c -arch arm64 -dynamiclib -miphoneos-version-min=11.0 -isysroot "#{sdk_path}" -o "#{app_framework_dylib}" -`
-  end
+  flutter_application_path ||= File.join('..', '..')
 
-  # Keep pod and script phase paths relative so they can be checked into source control.
-  # Process will be run from project directory.
+  export_script_directory = File.join(flutter_application_path, '.ios', 'Flutter')
 
-  # defined_in_file is set by CocoaPods and is a Pathname to the Podfile.
-  project_directory_pathname = defined_in_file.dirname
-  relative = current_directory_pathname.relative_path_from project_directory_pathname
-  pod '{{projectName}}', :path => relative.to_s, :inhibit_warnings => true
+  # Keep script phase paths relative so they can be checked into source control.
+  relative = flutter_relative_path_from_podfile(export_script_directory)
 
   flutter_export_environment_path = File.join('${SRCROOT}', relative, 'flutter_export_environment.sh');
+
+  # Compile App.framework and move it and Flutter.framework to "BUILT_PRODUCTS_DIR"
   script_phase :name => 'Run Flutter Build {{projectName}} Script',
     :script => "set -e\nset -u\nsource \"#{flutter_export_environment_path}\"\nexport VERBOSE_SCRIPT_LOGGING=1 && \"$FLUTTER_ROOT\"/packages/flutter_tools/bin/xcode_backend.sh build",
     :execution_position => :before_compile
+
+  # Embed App.framework AND Flutter.framework.
+  script_phase :name => 'Embed Flutter Build {{projectName}} Script',
+    :script => "set -e\nset -u\nsource \"#{flutter_export_environment_path}\"\nexport VERBOSE_SCRIPT_LOGGING=1 && \"$FLUTTER_ROOT\"/packages/flutter_tools/bin/xcode_backend.sh embed_and_thin",
```

Comment:
Instead of letting CocoaPods copy/re-codesign/strip the frameworks from the ephemeral directory, instead let the tool copy it via `embed_and_thin` after the rest of the app has compiled.  The tool already codesigned/stripped it during `assemble`.